vim-canvas-demo-app-react/src/components/settings-app/BrandCustomization.tsx

```diff
@@ -8,17 +8,29 @@ export const BrandCustomization: React.FC = () => {
   const [appColor, setAppColor] = useState<string>("#00FFE1");
   const { idToken } = useAuthTokenData();
   const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [saveButtonText, setSaveButtonText] = useState<string>("Save");
   const [isSaveButtonDisabled, setIsSaveButtonDisabled] =
     useState<boolean>(true);
 
   const handleSave = useCallback(() => {
-    saveSettings(
-      {
-        theme_color: appColor,
-      },
-      idToken
-    );
-    setIsSaveButtonDisabled(true);
+    const save = async () => {
+      try {
+        await saveSettings(
+          {
+            theme_color: appColor,
+          },
+          idToken
+        );
+        setTimeout(() => {
+          setSaveButtonText("Save");
+          setIsSaveButtonDisabled(true);
+        }, 2000);
+        setSaveButtonText("Saved!");
```

Comment:
If the user makes changes to the UI during the 2 seconds when the button says "saved!", when the timeout elapses the button will become disabled